src/coreComponents/constitutive/solid/DruckerPragerExtended.cpp

```diff
@@ -0,0 +1,174 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2019 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2019 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2019 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All right reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ *  @file DruckerPragerExtended.cpp
+ */
+
+#include "DruckerPragerExtended.hpp"
+
+namespace geosx
+{
+using namespace dataRepository;
+namespace constitutive
+{
+
+DruckerPragerExtended::DruckerPragerExtended( std::string const & name, Group * const parent ):
+  ElasticIsotropic( name, parent ),
+  m_defaultInitialFrictionAngle(),
+  m_defaultResidualFrictionAngle(),
+  m_defaultDilationRatio(),
+  m_defaultCohesion(),
+  m_defaultHardening(),
+  m_initialFriction(),
+  m_residualFriction(),
+  m_dilationRatio(),
+  m_pressureIntercept(),
+  m_hardening(),
+  m_newState(),
+  m_oldState()
+{
+  // register default values
+
+  registerWrapper( viewKeyStruct::defaultInitialFrictionAngleString, &m_defaultInitialFrictionAngle )->
+    setApplyDefaultValue( 30.0 )->
+    setInputFlag( InputFlags::OPTIONAL )->
+    setDescription( "Initial friction angle (degrees)" );
+
+  registerWrapper( viewKeyStruct::defaultResidualFrictionAngleString, &m_defaultResidualFrictionAngle )->
+    setApplyDefaultValue( 30.0 )->
+    setInputFlag( InputFlags::OPTIONAL )->
+    setDescription( "Residual friction angle (degrees)" );
+
+  registerWrapper( viewKeyStruct::defaultDilationRatioString, &m_defaultDilationRatio )->
+    setApplyDefaultValue( 1.0 )->
+    setInputFlag( InputFlags::OPTIONAL )->
+    setDescription( "Dilation ratio [0,1] (ratio = tan dilationAngle / tan frictionAngle)" );
+
+  registerWrapper( viewKeyStruct::defaultHardeningString, &m_defaultHardening )->
+    setApplyDefaultValue( 1.0e6 )->
+    setInputFlag( InputFlags::OPTIONAL )->
```

Comment:
Could you verify this value of a default 1e6 hardening parameter for the friction. I tested the model and I found some reasonable value in order of 1e-3 or 1e-8